pkg/server/handler/v1alpha1/integration.go

```diff
@@ -55,7 +57,35 @@ func ListIntegrations(ctx context.Context, tenant string, pagination *types.Pagi
 		integrations = append(integrations, *integration)
 	}
 
-	return types.NewListResponse(int(size), integrations[pagination.Start:end]), nil
+	results := []api.Integration{}
+	// Handle filter.
+	if qp.Filter == "" {
+		results = integrations[qp.Start:end]
+	} else {
+		filterParts := strings.Split(qp.Filter, "=")
+		if len(filterParts) != 2 {
+			return nil, cerr.ErrorUnknownInternal.Error("Filter query param %s is not correct", qp.Filter)
+		}
+		fieldsStr, keyWord := filterParts[0], filterParts[1]
+		fields := strings.Split(fieldsStr, ".")
+
+		for i := qp.Start; i < end; i++ {
+			item := integrations[i]
+			bytes, err := json.Marshal(item)
+			if err != nil {
+				return nil, err
+			}
+			value, err := jsonparser.GetString(bytes, fields...)
```

Comment:
It's better to use `jsonpath` instead of `jsonparser` for this purpose. It's much simpler and cyclone already have this vendor.   Refer to https://github.com/caicloud/cyclone/blob/master/pkg/workflow/workload/pod/secret.go#L97